common/web/input-processor/src/text/inputProcessor.ts

```diff
@@ -102,8 +102,9 @@ namespace com.keyman.text {
       }
 
       // Will handle keystroke-based non-layer change modifier & state keys, mapping them through the physical keyboard's version
-      // of state management.  `doModifierPress` must always run.
-      if(this.keyboardProcessor.doModifierPress(keyEvent, outputTarget, !fromOSK)) {
+      // of state management.  `doModifierPress` must always run for desktop-mode contexts.
+      // In standard Web integration, `contextDevice` indicates what type of OSK is visible.
+      if(!this.contextDevice.touchable && this.keyboardProcessor.doModifierPress(keyEvent, outputTarget, !fromOSK)) {
```

Comment:
I don't think this change is going to be 100% correct, because `doModifierPress` does "things" that aren't really modifier-related (those things should really be relocated -- but that may need to wait):

1. It clears deadkeys on backspace:

 
 
 https://github.com/keymanapp/keyman/blob/0212727b98e5555281162ea4b66fee20504004de/common/web/keyboard-processor/src/text/keyboardProcessor.ts#L727-L729

2. It sends notifications for modifier changes to the active keyboard (`KNS()`), which is used for example by Japanese and Korean IME-style keyboards (although these don't support touch, we don't know about other keyboards that may). We need to maintain the behaviour 'contract' for `KNS` even if it is not well documented.

There's other stuff going on: I don't fully follow the logic in the method -- I don't know what `Levent.LmodifierChange` is all about, nor what the `this.activeKeyboard.notify(0, outputTarget, 1)` really means.

Reply:
The issue is that this function _also_ pre-emptively sets the layer based on the physical keypress.  One or the other does need to be relocated, and it can't wait until after this PR.  The second point in the description cannot be properly addressed without it or something similar... or adding a new parameter or something.

Good call re: the deadkeys - that will be important.

> nor what the this.activeKeyboard.notify(0, outputTarget, 1) really means

That _is_ the `KNS` call.  Just a bit higher-level within the engine.

Reply:
Loosely, **K**eyman - **N**otify **S**tate.  That's what I inferred, anyway.

Reply:
... actually, a small change within the function might work.

Reply:
> > nor what the this.activeKeyboard.notify(0, outputTarget, 1) really means

> That is the KNS call. Just a bit higher-level within the engine.

Yes, I understood that. I just don't really know what the `0` value means in this context and why we are sending that notification to the keyboard. Doesn't help that `KNS` is undocumented!

![image](https://user-images.githubusercontent.com/4498365/177671348-22661239-af27-4b64-81b3-7cd06d5050bb.png)

Reply:
Oh, that.  Yeah, that's a decent question.  Reverse-engineering the CJK keyboards may reveal why, but there are probably bigger fish to fry right now.

Reply:
Agreed. It's on the backlog. 🐟🍳